### PR TITLE
Update AddInMemoryExporter extension method signature

### PR DESCRIPTION
**Pull request by utpilla**
## Changes

- Updated AddInMemoryExporter extension method to take in ICollection<Activity> for TracerProviderBuilder and ICollection<LogRecord> for OpenTelemetryLoggerOptions
- Updated InMemoryExporter<T> ctor to take in ICollection<T> as a parameter
- Removed ICollection<object> from InMemoryExporterOptions

### CODE REVIEW

File: src/OpenTelemetry.Exporter.InMemory/InMemoryExporterOptions.cs [hunk 1]
```diff
@@ -14,12 +14,9 @@
 // limitations under the License.
 // </copyright>
 
-using System.Collections.Generic;
-
 namespace OpenTelemetry.Exporter
 {
     public class InMemoryExporterOptions
     {
-        public ICollection<object> ExportedItems { get; set; }
```
Comment (alanwest) [inline src/OpenTelemetry.Exporter.InMemory/InMemoryExporterOptions.cs:23]: Can this whole class go away now?

Comment (utpilla) [reply src/OpenTelemetry.Exporter.InMemory/InMemoryExporterOptions.cs:23]: Yeah we could get rid of this class. I kept it just so if we add options for InMemoryExporter later on we could use this. We could always add the class later on as well.  Should I just delete it?

File: src/OpenTelemetry.Exporter.InMemory/README.md [hunk 2]
```diff
@@ -14,11 +14,11 @@ dotnet add package OpenTelemetry.Exporter.InMemory
...
-var list = new List<object>();
-var activityExporter = new InMemoryExporter<Activity>(
-    new InMemoryExporterOptions { ExportedItems = list });
-var logExporter = new InMemoryExporter<LogRecord>(
-    new InMemoryExporterOptions { ExportedItems = list });
+var activityList = new List<Activity>();
+var activityExporter = new InMemoryExporter<Activity>(activityList);
+
+var logRecordList = new List<LogRecord>();
+var logExporter = new InMemoryExporter<LogRecord>(logRecordList);
```
Comment (cijothomas) [inline src/OpenTelemetry.Exporter.InMemory/README.md:21]: we need to improve to an full usable example. Lets do it in a separate PR.

File: src/OpenTelemetry.Exporter.InMemory/InMemoryExporterOptions.cs (hunk 1 above)
Comment (cijothomas) [reply src/OpenTelemetry.Exporter.InMemory/InMemoryExporterOptions.cs:23]: yea deletre it. If we want to add any options, then add it then.

**Review by cijothomas — APPROVED** [review]

File: src/OpenTelemetry.Exporter.InMemory/InMemoryExporterOptions.cs (hunk 1 above)
Comment (utpilla) [reply src/OpenTelemetry.Exporter.InMemory/InMemoryExporterOptions.cs:23]: I have deleted this file.

**Review by alanwest — APPROVED** [review]

**Review by cijothomas — APPROVED** [review]

**Review by reyang — APPROVED** [review]
LGTM.
